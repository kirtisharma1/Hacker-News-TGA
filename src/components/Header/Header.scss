@import "../../styles/theme.scss";

.header {
  display: flex;
  align-items: center;
  background-color: $orangeColor;
  height: 2.4rem;
  margin-top: 1rem;

  &__logo {
    border: 0.1rem solid $whiteColor;
    margin: 0.2rem;
  }

  &__title {
    font-size: 1.4rem;
    color: $blackColor;
  }

  &__navigation {
    margin: 0 1rem;
  }

  &__menu-list {
    display: flex;
  }

  &__list-item {
    list-style: none;
    padding: 0.5rem;
    color: $blackColor;

    &--active,
    &--deactive {
      font-size: 1.3rem;
    }

    &--active {
      color: $whiteColor;
    }
  }
}
