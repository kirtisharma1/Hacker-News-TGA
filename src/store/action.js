import {
  UPDATE_FEED,
  UPDATE_PAGE,
  UPDATE_UPVOTED_LIST,
  UPDATE_HIDE_LIST,
  UPDATE_SORT_TYPE,
} from "./constants";

export const updateFeed = (data, dispatch) => {
  return dispatch({
    type: UPDATE_FEED,
    payload: data,
  });
};

export const updatePage = (data, dispatch) => {
  return dispatch({
    type: UPDATE_PAGE,
    payload: data,
  });
};

export const updateUpvotedList = (data, dispatch) => {
  return dispatch({
    type: UPDATE_UPVOTED_LIST,
    payload: data,
  });
};

export const updateHideList = (data, dispatch) => {
  return dispatch({
    type: UPDATE_HIDE_LIST,
    payload: data,
  });
};

export const updateSortType = (data, dispatch) => {
  return dispatch({
    type: UPDATE_SORT_TYPE,
    payload: data,
  });
};
