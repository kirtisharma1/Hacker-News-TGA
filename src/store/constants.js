export const UPDATE_FEED = "UPDATE_FEED";
export const UPDATE_SORT_TYPE = "UPDATE_SORT_TYPE";
export const UPDATE_UPVOTED_LIST = "UPDATE_UPVOTED_LIST";
export const UPDATE_HIDE_LIST = "UPDATE_HIDE_LIST";
export const UPDATE_PAGE = "UPDATE_PAGE";
export const HEADER_TITLE = "Hacker News";
export const HEADER_TOP_LABEL = "top";
export const HEADER_NEW_LABEL = "new";
export const HIDE_LABEL = "hide";
export const MORE_LABEL = "more";
export const LOADING_TEXT = "Loading...";
