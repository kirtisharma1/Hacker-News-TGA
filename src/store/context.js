import React from "react";

export const GlobalContext = React.createContext();

export const GlobalContextProvider = ({ data, children }) => (
  <GlobalContext.Provider value={data}>{children}</GlobalContext.Provider>
);
