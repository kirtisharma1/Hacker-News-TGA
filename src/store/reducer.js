import {
  UPDATE_FEED,
  UPDATE_PAGE,
  UPDATE_UPVOTED_LIST,
  UPDATE_HIDE_LIST,
  UPDATE_SORT_TYPE,
} from "./constants";
import { HEADER_TOP_LABEL } from "../constants";

export const initialState = {
  feed: [],
  sortType: HEADER_TOP_LABEL,
  upvotedList: [],
  hideList: [],
  page: 0,
};

export const GlobalReducer = (state, action) => {
  switch (action.type) {
    case UPDATE_FEED:
      return {
        ...state,
        feed: [...action.payload],
      };
    case UPDATE_PAGE:
      return {
        ...state,
        page: action.payload,
      };
    case UPDATE_UPVOTED_LIST:
      return {
        ...state,
        upvotedList: [...action.payload],
      };
    case UPDATE_HIDE_LIST:
      return {
        ...state,
        hideList: [...action.payload],
      };
    case UPDATE_SORT_TYPE:
      return {
        ...state,
        sortType: action.payload,
      };
    default:
      return state || initialState;
  }
};
