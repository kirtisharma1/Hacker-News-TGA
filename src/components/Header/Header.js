import React, { useContext } from "react";
// import logo from "../../static/logo.gif";
import { GlobalContext } from "../../store/context";
import {
  HEADER_NEW_LABEL,
  HEADER_TOP_LABEL,
  HEADER_TITLE,
} from "../../constants";
import { updateSortType } from "../../store/action";

export default function Header() {
  const { state, dispatch } = useContext(GlobalContext);
  const { sortType } = state;

  return (
    <header className="header container-wrapper">
      {/* <img src={logo} className="header__logo" alt="logo" /> */}
      <h1>{HEADER_TITLE}</h1>
      <nav className="header__navigation">
        <ul className="header__menu-list">
          <li className="header__list-item">
            <button
              type="button"
              className={`btn-empty header__list-item--${
                sortType === HEADER_TOP_LABEL ? "active" : "deactive"
              }`}
              onClick={() => updateSortType(HEADER_TOP_LABEL, dispatch)}
            >
              {HEADER_TOP_LABEL}
            </button>
          </li>
          <li className="header__list-item"> | </li>
          <li className="header__list-item">
            <button
              type="button"
              className={`btn-empty header__list-item--${
                sortType === HEADER_NEW_LABEL ? "active" : "deactive"
              }`}
              onClick={() => updateSortType(HEADER_NEW_LABEL, dispatch)}
            >
              {HEADER_NEW_LABEL}
            </button>
          </li>
        </ul>
      </nav>
    </header>
  );
}
