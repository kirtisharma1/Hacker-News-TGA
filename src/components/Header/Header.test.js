import React from "react";
import { render, fireEvent, wait } from "@testing-library/react";
import Header from "./Header";
import {
  HEADER_TITLE,
  HEADER_NEW_LABEL,
  HEADER_TOP_LABEL,
} from "../../constants";
import { GlobalContextProvider } from "../../store/context";
import { initialState } from "../../store/reducer";

const dispatch = jest.fn();
describe("Header component", () => {
  let result;
  beforeEach(() => {
    result = render(
      <GlobalContextProvider data={{ state: initialState, dispatch }}>
        <Header />
      </GlobalContextProvider>
    );
  });
  test("snapshot", () => {
    expect(result.asFragment()).toMatchSnapshot();
  });

  test("render without crash", () => {
    expect(result.getByText(HEADER_TITLE)).toBeInTheDocument();
  });

  test("menu item selection for 'new'", () => {
    const newMenuItem = result.getByText(HEADER_NEW_LABEL);
    fireEvent.click(newMenuItem);
  });

  test("menu item selection for 'new'", () => {
    result.rerender(
      <GlobalContextProvider
        data={{
          state: { ...initialState, sortType: HEADER_NEW_LABEL },
          dispatch: jest.fn(),
        }}
      >
        <Header />
      </GlobalContextProvider>
    );
    const newMenuItem = result.getByText(HEADER_TOP_LABEL);
    fireEvent.click(newMenuItem);
  });
});
