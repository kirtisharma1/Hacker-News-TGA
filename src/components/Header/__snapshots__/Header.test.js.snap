// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component snapshot 1`] = `
<DocumentFragment>
  <header
    class="header container-wrapper"
  >
    <h1>
      Hacker News
    </h1>
    <nav
      class="header__navigation"
    >
      <ul
        class="header__menu-list"
      >
        <li
          class="header__list-item"
        >
          <button
            class="btn-empty header__list-item--active"
            type="button"
          >
            top
          </button>
        </li>
        <li
          class="header__list-item"
        >
           | 
        </li>
        <li
          class="header__list-item"
        >
          <button
            class="btn-empty header__list-item--deactive"
            type="button"
          >
            new
          </button>
        </li>
      </ul>
    </nav>
  </header>
</DocumentFragment>
`;
